EasyGYM/UI/ViewModels/LibraryViewModel.swift

```diff
@@ -6,8 +6,111 @@
 //
 
 import Foundation
+import FirebaseAuth
+import FirebaseFirestore
 
 class LibraryViewModel {
     
-
+    var libraryRepo = WorkoutService()
+    var filteredWorkouts: [Workout] = []
+    var workouts: [Workout] = []
+    var isFiltering: Bool = false
+    var favoriteWorkoutIDs: Set<String> = []
+    
+    func filterWorkouts(by criteria: LibraryViewController.filterCriteria){
+        print("Filtering workouts by \(criteria)")
+        switch criteria {
+        case .push:
+            filteredWorkouts = workouts.filter { $0.movementType == .push }
+        case .pull:
+            filteredWorkouts = workouts.filter{ $0.movementType == .pull }
+        case .leg:
+            filteredWorkouts = workouts.filter{ $0.movementType == .leg }
+        }
+        print("Filtered Workouts: \(filteredWorkouts.count)")
+        isFiltering = true
+    }
+    
+    func clearFilter(){
+        isFiltering = false
+        filteredWorkouts.removeAll()
+    }
+    
+    func fetchWorkouts(completion: @escaping () -> Void) {
+        WorkoutService.shared.fetchWorkouts { [weak self] (workouts, error) in
+            if let error = error {
+                print("Error fetching workouts: \(error.localizedDescription)")
+                return
+            }
+            self?.workouts = workouts ?? []
+            print("Fetched Workouts: \(self?.workouts.count ?? 0)")
+            completion()
+        }
+    }
+    
+    func getWorkoutsCount() -> Int {
```

Comment:
this could be a computed property

var workoutsCount: Int {
   workouts.count
}